cli/azd/pkg/tools/bicep/bicep.go

```diff
@@ -5,170 +5,237 @@ package bicep
+	configDir, err := config.GetUserConfigDir()
+	if err != nil {
+		return "", err
+	}
 
-		cli.commandPath = commandPath
+	if runtime.GOOS == "windows" {
+		return filepath.Join(configDir, "bin", "bicep.exe"), nil
 	}
 
-	runArgs := exec.NewRunArgs(*cli.commandPath, args...)
-	return cli.commandRunner.Run(ctx, runArgs)
+	return filepath.Join(configDir, "bin", "bicep"), nil
 }
 
-const (
-	defaultBicepCommandPath string = "bicep"
-	envNameAzureConfigDir   string = "AZURE_CONFIG_DIR"
-)
+// downloadBicep downloads a given version of bicep from the release site, writing the output to name.
+func downloadBicep(ctx context.Context, transporter policy.Transporter, bicepVersion semver.Version, name string) error {
+	var releaseName string
+	switch runtime.GOOS {
+	case "windows":
+		releaseName = "bicep-win-x64.exe"
+	case "darwin":
+		releaseName = "bicep-osx-x64"
+	case "linux":
+		if _, err := os.Stat("/lib/ld-musl-x86_64.so.1"); err == nil {
+			releaseName = "bicep-linux-musl-x64"
+		} else {
+			releaseName = "bicep-linux-x64"
+		}
+	default:
+		return fmt.Errorf("unsupported platform")
+	}
+
+	bicepReleaseUrl := fmt.Sprintf("https://downloads.bicep.azure.com/v%s/%s", bicepVersion, releaseName)
 
-// Finds the bicep command path
-// Search in PATH, otherwise looks for standalone and az installation locations
-func findBicepPath() (*string, error) {
-	// First, check if tool is found in path
-	// This typically should return true for standalone install and false for `az` bundled install
-	found, err := tools.ToolInPath(defaultBicepCommandPath)
+	log.Printf("downloading bicep release %s -> %s", bicepReleaseUrl, name)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", bicepReleaseUrl, nil)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	if found {
-		return convert.RefOf(defaultBicepCommandPath), nil
+	resp, err := transporter.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("http error %d", resp.StatusCode)
 	}
 
-	// If not found in path, check in 2 locations
-	// Default location for standalone install
-	// Default location for az bicep install
-	user, err := user.Current()
+	f, err := os.CreateTemp(filepath.Dir(name), fmt.Sprintf("%s.tmp*", filepath.Base(name)))
 	if err != nil {
-		return nil, fmt.Errorf("failed getting current user: %w", err)
+		return err
 	}
+	defer func() {
+		_ = f.Close()
+		_ = os.Remove(f.Name())
+	}()
 
-	var bicepFilename string
-	var bicepStandalonePath string
+	if _, err := io.Copy(f, resp.Body); err != nil {
+		return err
+	}
 
-	if runtime.GOOS == "windows" {
-		bicepFilename = "bicep.exe"
-		bicepStandalonePath = filepath.Join(user.HomeDir, fmt.Sprintf("AppData\\Local\\Programs\\Bicep CLI\\%s", bicepFilename))
-	} else {
-		bicepFilename = "bicep"
-		bicepStandalonePath = fmt.Sprintf("/usr/local/bin/%s", bicepFilename)
+	if err := f.Chmod(osutil.PermissionExecutableFile); err != nil {
+		return err
+	}
```

Comment:
Should we perform the chmod after the move/rename or does the order not matter?

Reply:
The order doesn't matter (the permission flows with the file).